### ISSUE TITLE
Export-Store: collection name mismatch in collection structure file

### ISSUE THREAD

**Issue by aurelijusb** [arangodb/arangodb#1556]
tl;dr
-----
Cannot export `2.7.0` stable into `2.7.0-devel` (`b2bc61d02d8f335cb8395eba97026b40e17405fb`) using `arangodump` and `arangorestore`

Context
----------
 2.7.0 has bug in Web Graph visualisation. Wanted to check if it is fixed in develop version.

Steps to reproduce
-------------------------

* Store old data into `/var/lib/arangodb/databases`:  https://www.dropbox.com/s/w53nalad6ciwwh4/database-150862994665.tar.gz?dl=0
* Run 2.7.0 on port 8530:
```
#!/bin/sh

SBIN_DIR=/usr/sbin
JS_DIR=/usr/share/arangodb
DATA_DIR=/var/lib/arangodb
ENDPOINT=127.0.0.1:8530
LOGS_DIR=/home/aurelijus/Documents/ArangoDBTests/INSTANCE3

$SBIN_DIR/arangod $DATA_DIR --configuration none --server.endpoint "tcp://$ENDPOINT" --javascript.startup-directory $JS_DIR/js --javascript.app-path $DATA_DIR/app --log.file $LOGS_DIR/log
```

* Run 2.7.0 develop on port 8529
```
#!/bin/sh
BIN_DIR=/some/directory/where/I/cloned/git/repository/arangodb
DATA_DIR=/some/directory/where/I/wanted/to/dump/data
ENDPOINT=127.0.0.1:8529

$BIN_DIR/bin/arangod $DATA_DIR --configuration none --server.endpoint "tcp://$ENDPOINT" --javascript.startup-directory $BIN_DIR/js --javascript.app-path $DATA_DIR/app --log.file $DATA_DIR/log
```
* Export from old database: 
```
 arangodump --log.level debug --server.endpoint="tcp://127.0.0.1:8530" --server.database=multimodel --server.username=root
```
```
2015-11-02T19:08:51Z [20556] DEBUG [arangosh/ArangoShell/ArangoClient.cpp:393] using init file '/etc/arangodb/arangodump.conf'
Please specify a password: 
2015-11-02T19:09:02Z [20556] DEBUG [lib/Rest/EndpointIp.cpp:353] connecting to ip endpoint 'tcp://127.0.0.1:8530'
Server version: 2.7.0
Connected to ArangoDB 'tcp://127.0.0.1:8530', database: 'multimodel', username: 'root'
Writing dump to output directory '/home/aurelijus/Documents/ArangoDBTests/INSTANCE3/dump'
Last tick provided by server is: 160285629927
dumping collection 'Database'...
dumping collection 'Type'...
dumping collection 'ExampleOf'...
Processed 3 collection(s), wrote 20826 byte(s) into datafiles, sent 3 batch(es)
```
Password of old database: `test321`
* Files are saved: https://www.dropbox.com/s/u06t7r18n38qa2k/dump.tar.gz?dl=0
* Trying to import fails:
```
/some/directory/where/I/cloned/git/repository/arangodb/bin/arangorestore --log.level debug --server.endpoint="tcp://127.0.0.1:8529" --input-directory "dump"
```
```
2015-11-02T19:09:32Z [20589] DEBUG [arangosh/ArangoShell/ArangoClient.cpp:375] no system init file '/usr/local/etc/arangodb/arangorestore.conf'
Please specify a password: 
2015-11-02T19:09:34Z [20589] DEBUG [lib/Rest/EndpointIp.cpp:353] connecting to ip endpoint 'tcp://127.0.0.1:8529'
Server version: 2.7.0-devel
Connected to ArangoDB 'tcp://127.0.0.1:8529
Processed 0 collection(s), read 0 byte(s) from datafiles, sent 0 batch(es)
collection name mismatch in collection structure file 'Type_' (offending value: 'Type')
```

Environment
----------------
```
arangod --version
architecture: 64bit
build-date: 2015-10-13 22:17:34
configure:  './configure' 'CC=gcc' 'CXX=g++' 'CFLAGS=-g -O3 -fomit-frame-pointer' 'CXXFLAGS=-g -O3 -fomit-frame-pointer' '--enable-gold' '--enable-tcmalloc' '--program-prefix=' '--prefix=/usr' '--exec-prefix=/usr' '--bindir=/usr/bin' '--sbindir=/usr/sbin' '--sysconfdir=/etc' '--datadir=/usr/share' '--includedir=/usr/include' '--libdir=/usr/lib' '--libexecdir=/usr/lib' '--localstatedir=/var' '--sharedstatedir=/usr/com' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--enable-all-in-one-etcd' '--enable-internal-go'
env:  CFLAGS='-g -O3 -fomit-frame-pointer' CXXFLAGS='-g -O3 -fomit-frame-pointer' CC='gcc' CXX='g++'
fd-client-event-handler: poll
fd-setsize: 1024
icu-version: 54.1
libev-version: 4.11
maintainer-mode: false
openssl-version: OpenSSL 1.0.1f 6 Jan 2014
readline-version: 6.3
server-version: 2.7.0
sizeof int: 4
sizeof void*: 8
tcmalloc: true
v8-version: 4.3.61
```
```
/some/directory/where/I/cloned/git/repository/arangodb/bin/arangod --version
architecture: 64bit
build-date: 2015-11-02 19:00:42
configure:  './configure'
fd-client-event-handler: poll
fd-setsize: 1024
icu-version: 54.1
libev-version: 4.11
maintainer-mode: false
openssl-version: OpenSSL 1.0.2d 9 Jul 2015
readline-version: unknown version
repository-version: heads/devel-0-gb2bc61d02d8f335cb8395eba97026b40e17405fb
server-version: 2.7.0-devel
sizeof int: 4
sizeof void*: 8
tcmalloc: false
v8-version: 4.3.61
```
OS: Ubuntu 15.10

Workaround
----------------

Hard copy folder from `/var/lib/arangodb/databases` to new database data directory.

**Comment by dothebart** [comment]
You're probably missing the graph definitions from your import?
you therefore also have to dump/restore the system databases (starting with an underscore) since they contain the graph definition (the _graphs collection)
This can be achieved by 
`--include-system-collections true`

**Comment by aurelijusb** [comment]
Error remains:
```
arangodump --log.level debug --server.endpoint="tcp://127.0.0.1:8530" --server.database=multimodel --include-system-collections true --server.username=root
2015-11-03T17:13:24Z [7358] DEBUG [arangosh/ArangoShell/ArangoClient.cpp:393] using init file '/etc/arangodb/arangodump.conf'
Please specify a password: 
2015-11-03T17:13:28Z [7358] DEBUG [lib/Rest/EndpointIp.cpp:353] connecting to ip endpoint 'tcp://127.0.0.1:8530'
Server version: 2.7.0
Connected to ArangoDB 'tcp://127.0.0.1:8530', database: 'multimodel', username: 'root'
Writing dump to output directory '/home/aurelijus/Documents/ArangoDBTests/INSTANCE4/dump'
Last tick provided by server is: 161610081880
dumping collection '_apps'...
dumping collection '_aqlfunctions'...
dumping collection '_graphs'...
dumping collection '_modules'...
dumping collection '_routing'...
dumping collection '_sessions'...
dumping collection '_system_users_users'...
dumping collection '_users'...
dumping collection 'Database'...
dumping collection 'Type'...
dumping collection 'ExampleOf'...
Processed 11 collection(s), wrote 92342 byte(s) into datafiles, sent 11 batch(es)
aurelijus@aurelijus-aspire:~/Documents/ArangoDBTests/INSTANCE4$ /home/aurelijus/Documents/OPEN-SOURCE/arangodb/bin/arangorestore --log.level debug --server.endpoint="tcp://127.0.0.1:8529" --input-directory "dump"
2015-11-03T17:13:57Z [7363] DEBUG [arangosh/ArangoShell/ArangoClient.cpp:375] no system init file '/usr/local/etc/arangodb/arangorestore.conf'
Please specify a password: 
2015-11-03T17:13:59Z [7363] DEBUG [lib/Rest/EndpointIp.cpp:353] connecting to ip endpoint 'tcp://127.0.0.1:8529'
Server version: 2.7.0-devel
Connected to ArangoDB 'tcp://127.0.0.1:8529
Processed 0 collection(s), read 0 byte(s) from datafiles, sent 0 batch(es)
collection name mismatch in collection structure file 'Type_' (offending value: 'Type')
```
Generated files: https://www.dropbox.com/s/3rgr5r76k4v21dm/dump-include-system-collections.tar.gz?dl=0

**Comment by jsteemann** [comment]
I found the reason for this. You mentioned that your 2.7.0-devel branch is on commit `b2bc61d02d8f335cb8395eba97026b40e17405fb`. According to `git log`, this commit is from Wed Aug 19 14:46:28 2015 +0200.

Apart from that this version is more than two months old (quite old for a `devel`), this exact version seems to have an issue with the dump/restore filenames. The dump/restore filenames were changed at Aug 17, and there was a bugfix for the names on Aug 20. 

Here are the commits affecting the dump/restore filenames:
```
commit 47c7a873b27261fa4f24c7ba588f693503ce3f67
Date:   Thu Aug 20 02:47:00 2015 +0200

    fix dump/restore filenames

commit 651ccac94d52420403c1630e2817f29b54971315
Date:   Mon Aug 17 17:31:15 2015 +0200

    set the mac to add hashes too.

commit 29d2285f69d075745f8d52224d09fc9fb47d3f1b
Date:   Mon Aug 17 17:08:06 2015 +0200

    Adjust the import site of the dumper for hashes in the filenames.
```
So it seems that you're on a commit that contained the filename changes for dump/restore, but still contains a bug for them. Going a few days ahead to the commit from Aug 20 will fix that, or even better, moving to the current state of `devel`.
By the way, there has been a repository cleanup after Aug 2015, which means the commit repository has been rewritten. If you want to update your devel to a recent commit, I suggest using recloning or `git reset --hard origin/devel`.
